### PR TITLE
Solution to mini-challenges-4 labs

### PR DESCRIPTION
**Pull request by AdefabiAdekunle**

### CODE REVIEW

File: src/word-search/index.js [hunk 1]
```diff
@@ -1,3 +1,37 @@
-function wordSearch(words, word) {}
+function wordSearch(words, word) {
+
+  function dfs(x, y, index) {
+    if (x < 0 || x > words.length-1  || y < 0 || y > words[0].length) return false;
+    if (words[x][y] !== word[index]) return false;
+    if (index === word.length-1) return true;
+    var orignalCharacter = words[x][y];
+
```
Comment (flinchy) [inline src/word-search/index.js:8]: prefer let to var

File: src/word-search/index.js [hunk 2]
```diff
@@ -1,3 +1,37 @@
-function wordSearch(words, word) {}
+function wordSearch(words, word) {
+
+  function dfs(x, y, index) {
+    if (x < 0 || x > words.length-1  || y < 0 || y > words[0].length) return false;
+    if (words[x][y] !== word[index]) return false;
+    if (index === word.length-1) return true;
+    var orignalCharacter = words[x][y];
```
Comment (flinchy) [inline src/word-search/index.js:7]: prefer let to var

File: src/word-search/index.js [hunk 3]
```diff
@@ -1,3 +1,37 @@
-function wordSearch(words, word) {}
+function wordSearch(words, word) {
+
+  function dfs(x, y, index) {
```
Comment (flinchy) [inline src/word-search/index.js:3]: why not move the dfs function out of the Wordsearch function

File: src/count-islands/index.js [hunk 4]
```diff
@@ -1,3 +1,43 @@
...
+    //visit the neigbouring land in four corner direction
+    deepFirstSearch(index_i - 1, index_j); //Up
+    deepFirstSearch(index_i, index_j - 1); //LEFT
+    deepFirstSearch(index_i + 1, index_j); //Down
+    deepFirstSearch(index_i, index_j + 1); //Right
+
+  }
+
+  let noOfIsland = 0;
+  for (var i = 0; i < nRows; i++) {
```
Comment (flinchy) [inline src/count-islands/index.js:28]: always prefer let to var

File: src/count-islands/index.js [hunk 5]
```diff
@@ -1,3 +1,43 @@
-function countIslands(grid) {}
+function countIslands(grid) {
+  let nRows = grid.length;
+  let nCols = grid[0].length;
+
+  function deepFirstSearch(index_i, index_j) {
```
Comment (flinchy) [inline src/count-islands/index.js:5]: move the deepFirstSearch() function out of the countIsLands() function

File: src/word-search/index.js [hunk 6]
```diff
@@ -1,3 +1,37 @@
...
+      dfs(x, y + 1, index + 1) == true//right
+    ) {
+      return true;
+    }
+    words[x][y] = orignalCharacter; // backtracking
+    return false;
+  }
+
+  let booleanArrayOfWordFoundUsingWordFirstLetter = [];
+  for (var row = 0; row < words.length; row++) {
```
Comment (flinchy) [inline src/word-search/index.js:24]: always prefer let to var

**Comment by AdefabiAdekunle** [comment]
Thanks Chisom . I have always thought  using var inside the for loop
parentheses doesn’t matter. But have learnt a new thing now.
Thanks

**Pull request merged**